include/hc_printf.hpp

```diff
@@ -285,7 +329,13 @@ static inline PrintfError printf(const char* format_string, All... all) [[hc,cpu
 #endif
 
 // regex for finding format string specifiers
-static const std::regex specifierPattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}([diuoxXfFeEgGaAcsp]){1}");
+static const std::regex specifierPattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}([hl]*)([diuoxXfFeEgGaAcsp]){1}");
+static const std::regex signedInt16Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(h)([di]){1}");
+static const std::regex unsignedInt16Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(h)([uoxX]){1}");
+static const std::regex signedInt32Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(l)([di]){1}");
+static const std::regex unsignedInt32Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(l)([uoxX]){1}");
+static const std::regex signedInt64Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(ll)([di]){1}");
+static const std::regex unsignedInt64Pattern("(%){1}[-+#0]*[0-9]*((.)[0-9]+){0,1}(ll)([uoxX]){1}");
```

Comment:
We don't need all these new regex patterns, we probably just need to add ([hl]*) to the existing signed/unsigned integer patterns